app/views/posts/_form.html.erb

```diff
@@ -0,0 +1,28 @@
+<%= form_for @post do |f| %>
+
+<% if @post.errors.any? %>
+    <div id="error_explanation">
+      <h2>
+        <%= pluralize(@post.errors.count, "error") %> prohibited
+        this article from being saved:
+      </h2>
+      <ul>
+       <% @post.errors.full_messages.each do |msg| %>
+         <li><%= msg %></li>
+       <% end %>
+     </ul>
+  </div>
+  <% end %>
```

Comment:
👍 nice error handling, watch your indentation  `<% end %>` should match the indentation with the `<% if ...`